neelance/errors/errors.go

```diff
@@ -49,25 +51,35 @@ func (err *QueryError) Error() string {
 
 var _ error = &QueryError{}
 
+// FormattedError a formatted error which includes the error message and extra information
+// which is JSON encoded and passed with the error.
+type FormattedError struct {
+	Message string
+	Extra   interface{}
+}
+
+// ErrorMessageFunc a func which given an error returns a formatted error.
+type ErrorMessageFunc func(err error) FormattedError
```

Comment:
I don't think this is quite right. It only lets the user change whats in this "extra" key.

Maybe it could return a graphql.Marshaler?